pandas/core/internals/blocks.py

```diff
@@ -2815,12 +2814,11 @@ def concat_same_type(self, to_concat, placement=None):
         1. Change Categorical._concat_same_type to use union_categoricals
         2. Delete this method.
         """
-        values = self._concatenator(
-            [blk.values for blk in to_concat], axis=self.ndim - 1
-        )
+        values = self._holder._concat_same_type([blk.values for blk in to_concat])
```

Comment:
I know the `_concatenator` -> `_holder._concat_same_type` change is right now equivalent. But I would maybe leave it as is for now, because actually `Categorical._concat_same_type` is not implementing the spec of returning its own type / this problem is something we need to fix in general for ExtensionArrays.

Reply:
Either way I guess.  I'm hopeful Block.concat_same_type will be gone by then

Reply:
reverted